internal/k8s/validation.go

```diff
@@ -12,86 +12,218 @@ import (
+	sslRedirectAnnotation                 = "ingress.kubernetes.io/ssl-redirect"
+	proxyBufferingAnnotation              = "nginx.org/proxy-buffering"
+	hstsAnnotation                        = "nginx.org/hsts"
+	hstsMaxAgeAnnotation                  = "nginx.org/hsts-max-age"
+	hstsIncludeSubdomainsAnnotation       = "nginx.org/hsts-include-subdomains"
+	hstsBehindProxyAnnotation             = "nginx.org/hsts-behind-proxy"
+	proxyBuffersAnnotation                = "nginx.org/proxy-buffers"
+	proxyBufferSizeAnnotation             = "nginx.org/proxy-buffer-size"
+	proxyMaxTempFileSizeAnnotation        = "nginx.org/proxy-max-temp-file-size"
+	upstreamZoneSizeAnnotation            = "nginx.org/upstream-zone-size"
+	jwtRealmAnnotation                    = "nginx.com/jwt-realm"
+	jwtKeyAnnotation                      = "nginx.com/jwt-key"
+	jwtTokenAnnotation                    = "nginx.com/jwt-token"
+	jwtLoginURLAnnotation                 = "nginx.com/jwt-login-url"
+	listenPortsAnnotation                 = "nginx.org/listen-ports"
+	listenPortsSSLAnnotation              = "nginx.org/listen-ports-ssl"
+	keepaliveAnnotation                   = "nginx.org/keepalive"
+	maxFailsAnnotation                    = "nginx.org/max-fails"
+	maxConnsAnnotation                    = "nginx.org/max-conns"
+	failTimeoutAnnotation                 = "nginx.org/fail-timeout"
+	appProtectEnableAnnotation            = "appprotect.f5.com/app-protect-enable"
+	appProtectSecurityLogEnableAnnotation = "appprotect.f5.com/app-protect-security-log-enable"
+	internalRouteAnnotation               = "nsm.nginx.com/internal-route"
+	websocketServicesAnnotation           = "nginx.org/websocket-services"
+	sslServicesAnnotation                 = "nginx.org/ssl-services"
+	grpcServicesAnnotation                = "nginx.org/grpc-services"
+	rewritesAnnotation                    = "nginx.org/rewrites"
+	stickyCookieServicesAnnotation        = "nginx.com/sticky-cookie-services"
 )
 
 type annotationValidationContext struct {
-	annotations map[string]string
-	name        string
-	value       string
-	isPlus      bool
-	fieldPath   *field.Path
+	annotations           map[string]string
+	name                  string
+	value                 string
+	isPlus                bool
+	appProtectEnabled     bool
+	internalRoutesEnabled bool
+	fieldPath             *field.Path
 }
 
 type annotationValidationFunc func(context *annotationValidationContext) field.ErrorList
 type annotationValidationConfig map[string][]annotationValidationFunc
 type validatorFunc func(val string) error
 
 var (
-	// nginxAnnotationValidations defines the various validations which will be applied in order to each ingress
-	// annotation for nginx. If any specified validation fails, the remaining validations for that annotation will not
-	// be run.
-	nginxAnnotationValidations = annotationValidationConfig{
+	// annotationValidations defines the various validations which will be applied in order to each ingress annotation.
```

Comment:
I backed away from the idea of defining separate sets of validation depending on whether this is Nginx or Nginx Plus, it just complicated things.